bottlerocket/agents/src/bin/ecs-workload-agent/main.rs

```diff
@@ -0,0 +1,499 @@
+            config: spec.configuration,
+            aws_secret_name: spec.secrets.get(AWS_CREDENTIALS_SECRET_NAME).cloned(),
+        })
+    }
+
+    /// Run the set of configured workload tests and return the results.
+    async fn run(&mut self, _info_client: &I) -> Result<TestResults, Self::E> {
+        let config = aws_config(
+            &self.aws_secret_name.as_ref(),
+            &self.config.assume_role,
+            &None,
+            &self.config.region,
+            false,
+        )
+        .await?;
+        let ecs_client = aws_sdk_ecs::Client::new(&config);
+
+        info!("Waiting for registered container instances...");
+
+        tokio::time::timeout(
+            Duration::from_secs(30),
+            wait_for_cluster_ready(&ecs_client, &self.config.cluster_name),
+        )
+        .await
+        .context(error::InstanceTimeoutSnafu)??;
+
+        // First loop through and make sure we have task definitions for each of our tests.
+        // We don't want to starts tasks if one of the defined tets is not able to be run.
+        let mut task_def_arns = Vec::new();
+        for test_def in &self.config.tests {
+            let task_def_arn = create_or_find_task_definition(&ecs_client, test_def).await?;
+
+            task_def_arns.push(task_def_arn);
+        }
+
+        // Now that we have the full set of task definitions to run, loop through and kick them all off.
+        let mut task_arns = Vec::new();
+        for task_def_arn in task_def_arns {
+            info!("Running task '{}'", task_def_arn);
+            let run_task_output = ecs_client
+                .run_task()
+                .cluster(&self.config.cluster_name)
+                .task_definition(task_def_arn)
+                .launch_type(LaunchType::Ec2)
+                .send()
+                .await
+                .context(error::TaskRunCreationSnafu)?;
+            let run_task_arns: Vec<String> = run_task_output
+                .tasks()
+                .map(|tasks| {
+                    tasks
+                        .iter()
+                        .filter_map(|task| task.task_arn().map(|arn| arn.to_string()))
+                        .collect()
+                })
+                .unwrap();
+            for task_arn in &run_task_arns {
+                task_arns.push(task_arn.clone());
+            }
+        }
+
+        info!("Waiting for tasks to complete...");
+
+        match tokio::time::timeout(
+            Duration::from_secs(30),
+            wait_for_test_completion(&ecs_client, &self.config.cluster_name, &task_arns),
+        )
+        .await
+        {
+            Ok(results) => results,
+            Err(_) => test_results(&ecs_client, &self.config.cluster_name, &task_arns).await,
+        }
+    }
+
+    /// Terminate the test runs. This is a noop for this agent.
+    async fn terminate(&mut self) -> Result<(), Self::E> {
+        Ok(())
+    }
+}
+
+/// Loops waiting for the workload test tasks to complete.
```

Comment:
Should this send updates to the crd like sonobuoy does now?

Reply:
Can you point me to where things are updated in the latest? I'm missing that part. I don't see what changed in that respect in https://github.com/bottlerocket-os/bottlerocket-test-system/pull/690/files#diff-bb5e446ed77b2db083260a5cd9e88200daf3f723eec7d88d9ba1840f7b98fd8d

Reply:
https://github.com/bottlerocket-os/bottlerocket-test-system/blob/1e8122cb99c4b45bc2b58730bb1b50d9d6da1369/bottlerocket/agents/src/sonobuoy.rs#L26-L36

Reply:
I think I've got it. Please take a look at the [diff from the last push](https://github.com/bottlerocket-os/bottlerocket-test-system/compare/1ca83cb85952a2b6c46b24a5426d15d3fbb62913..8dadb103076282231e9d7896b6fff0ccf8c19749) and let me know if that is not right.

Reply:
Looks good.